src/Data/Artifacts/PaleFlame/index.tsx

```diff
@@ -1,23 +1,34 @@
 import icons from './icons'
-import { Data } from '../../../Formula/type'
-import { lookup, naught, percent, greaterEq } from '../../../Formula/utils'
+import { Data, Info } from '../../../Formula/type'
+import { lookup, naught, percent, greaterEq, sum, equal } from '../../../Formula/utils'
 import { input } from '../../../Formula'
 import { ArtifactSetKey } from '../../../Types/consts'
 import { ArtifactSheet, IArtifactSheet } from '../ArtifactSheet'
 import { dataObjForArtifactSheet } from '../dataUtil'
 import { cond, sgt, st } from '../../SheetUtil'
+import { range } from '../../../Util/Util'
 
 const key: ArtifactSetKey = "PaleFlame"
 
-const [condStatePath, condState] = cond(key, "state")
+const [condStackPath, condStack] = cond(key, "stacks")
 
-const set2 = greaterEq(input.artSet.PaleFlame, 2, lookup(condState, { "twostacks": percent(0.5) }, percent(0.25)))
-const set4 = greaterEq(input.artSet.PaleFlame, 4, lookup(condState, { "onestack": percent(0.09), "twostacks": percent(0.18) }, naught))
+const physical_dmg_info: Info = { key: "physical_dmg_" }
+const set2 = greaterEq(input.artSet.PaleFlame, 2, percent(0.25), physical_dmg_info)
```

Comment:
This set2 effect was working fine before, why change it to be more complicated?

Reply:
The previous one changes the 2 set effect to 50% if the condition is at two stacks. It might confuse users when just looking at the 2 set effect I guess? Refer to images here: https://imgur.com/a/3VsaNe9
But if you think its fine then I guess its fine.

Reply:
I think the existing implementation is fine. its rarely the case that only the 4set is displayed.

Reply:
Don't you need to guard "twostack" behind C4? It could linger after PF4 are removed. The old code doesn't seem to do that